Src/GUI/Plugins/PlotterPlugin.py

```diff
@@ -469,28 +472,40 @@ def on_load_data(self, event=None):
         filelist = dialog.GetPaths()
         
         for basename, filename in zip(baselist, filelist):
-            with netCDF4.Dataset(filename, "r") as f:
-                _vars = f.variables
-                data = collections.OrderedDict()
-                for k in _vars:
-                    dtype = _vars[k][:].dtype
-                    if not numpy.issubdtype(dtype, numpy.number):
-                        continue
-                    data[k]={}
-                    if hasattr(_vars[k], 'axis'):
-                        if _vars[k].axis:
-                            data[k]['axis'] = _vars[k].axis.split('|')
-                        else:
-                            data[k]['axis'] = []
+
+            ext = os.path.splitext(filename)[1]
+            for cls, exts in PLOTTER_DATA_TYPES.items():
+                if ext in exts:
+                    f = cls(filename,"r")
+                    break
+            else:
+                continue
+
+            data = collections.OrderedDict()
+            for vname, vinfo in f.variables.items():
+                vpath, variable = vinfo
+                arr = variable.get_array()
+
+                if not numpy.issubdtype(arr.dtype, numpy.number):
+                    continue
+
+                data[vname] = {}
+                if hasattr(variable, 'axis'):
+                    axis = getattr(variable,'axis')
+                    if axis:
+                        data[vname]['axis'] = axis.split('|')
                     else:
-                        data[k]['axis'] = []
-                    data[k]['data'] = _vars[k][:]
-                    data[k]['units'] = getattr(_vars[k], "units", "au")
-            
-                unique_name = self.unique(basename, self.plugin._dataDict)
-            
-                self.plugin._dataDict[unique_name]={'data': data, 'path': filename, 'basename': basename}
-                self.plugin._dataPanel.show_dataset(-1)
+                        data[vname]['axis'] = []
+                else:
+                    data[vname]['axis'] = []
+                data[vname]['path'] = vpath
+                data[vname]['data'] = arr
+                data[vname]['units'] = getattr(variable, "units", "au")
+
+            unique_name = self.unique(basename, self.plugin._dataDict)
+        
+            self.plugin._dataDict[unique_name]={'data': data, 'path': filename, 'basename': basename}
+            self.plugin._dataPanel.show_dataset(-1)
```

Comment:
I can't see the netcdf file being closed; am I wrong? Ideally, I would love if we could implement a context manager to handle the file, to ensure that it is always closed.

Reply:
yeap a context manager would the best thing to do